db/appointments.ts

```diff
@@ -17,6 +19,32 @@ const getAll = async (psychologistId: string): Promise<AppointmentWithPatient[]>
   }
 };
 
+const getLastWeekByUniversity = async (): Promise<{name: string, count: string}[]> => {
+  try {
+    const now = new Date();
+    const lastWeek = new Date();
+    now.setHours(18, 0, 0, 0);
+    lastWeek.setHours(18, 0, 0, 0);
+    lastWeek.setDate(now.getDate() - 7);
+
+    const appointments = await db
+    .select(`${universitiesTable}.name`)
+    .from(appointmentsTable)
+    .innerJoin(psychologistsTable, `${appointmentsTable}.psychologistId`, `${psychologistsTable}.dossierNumber`)
+    .innerJoin(universitiesTable, `${psychologistsTable}.assignedUniversityId`, `${universitiesTable}.id`)
+    .where(`${appointmentsTable}.appointmentDate`, '>=', lastWeek.toISOString())
+    .where(`${appointmentsTable}.appointmentDate`, '<', now.toISOString())
+    .count('*')
+    .groupBy(`${universitiesTable}.name`)
+    .orderBy(`${universitiesTable}.name`);
```

Comment:
I guess this may return a number of appointments for '-- Aucune pour le moment' university. Not sure this is a problem but I just wanted to point it out.